api/tests/opentrons/protocol_runner/test_task_queue.py

```diff
@@ -1,94 +1,77 @@
+    subject = TaskQueue()
+    subject.set_run_func(func=run_func)
     subject.start()
-    assert subject.is_started() is True
+    await subject.join()
 
+    decoy.verify(await run_func())
 
-async def test_add_run_task() -> None:
-    """It should be able to add a task for the "run" phase."""
-    run_result = False
 
-    async def run_task() -> None:
-        nonlocal run_result
-        run_result = True
+async def test_set_cleanup_func(decoy: Decoy) -> None:
+    """It should be able to add a task for the "cleanup" phase."""
+    cleanup_func = decoy.mock(is_async=True)
 
     subject = TaskQueue()
-    subject.add(phase=TaskQueuePhase.RUN, func=run_task)
+    subject.set_cleanup_func(func=cleanup_func)
     subject.start()
     await subject.join()
 
-    assert run_result is True
+    decoy.verify(await cleanup_func(error=None))
 
 
-async def test_add_cleanup_task() -> None:
-    """It should be able to add a task for the "cleanup" phase."""
-    cleanup_result = False
-
-    async def cleanup_task() -> None:
-        nonlocal cleanup_result
-        cleanup_result = True
+async def test_passes_args(decoy: Decoy) -> None:
+    """It should pass kwargs to the run phase function."""
+    run_func = decoy.mock(is_async=True)
 
     subject = TaskQueue()
-    subject.add(phase=TaskQueuePhase.CLEANUP, func=cleanup_task)
+    subject.set_run_func(func=run_func, hello="world")
     subject.start()
     await subject.join()
 
-    assert cleanup_result is True
-
+    decoy.verify(await run_func(hello="world"))
 
-async def test_passes_args() -> None:
-    """It should be able to add a task for the "cleanup" phase."""
-    run_args = None
-    cleanup_args = None
-
-    async def run_task(*args: Any, **kwargs: Any) -> None:
-        nonlocal run_args
-        run_args = (args, kwargs)
 
-    async def cleanup_task(*args: Any, **kwargs: Any) -> None:
-        nonlocal cleanup_args
-        cleanup_args = (args, kwargs)
+async def test_cleanup_runs_second(decoy: Decoy) -> None:
+    """It should run the "run" and "cleanup" funcs in order."""
+    run_func = decoy.mock(is_async=True)
+    cleanup_func = decoy.mock(is_async=True)
 
     subject = TaskQueue()
-    subject.add(TaskQueuePhase.RUN, run_task, 1, 2, hello="world")
-    subject.add(TaskQueuePhase.CLEANUP, cleanup_task, "a", "b", fizz="buzz")
+    subject.set_run_func(func=run_func)
+    subject.set_cleanup_func(func=cleanup_func)
     subject.start()
     await subject.join()
 
-    assert run_args == ((1, 2), {"hello": "world"})
-    assert cleanup_args == (("a", "b"), {"fizz": "buzz"})
+    decoy.verify(
+        await run_func(),
+        await cleanup_func(error=None),
+    )
 
 
-async def test_cleanup_always_runs() -> None:
-    """It should be able to add a task for the "cleanup" phase."""
-    cleanup_result = False
+async def test_cleanup_gets_run_error(decoy: Decoy) -> None:
+    """It should verify "cleanup" func gets error raised in "run" func."
```

Comment:
Oh shoot! Sorry, the quotes got messed up!
```suggestion
    """It should verify "cleanup" func gets error raised in "run" func."""
```